### PR TITLE
Strongly typed id module

### PR DESCRIPTION
**Pull request by mithileshz**

### CODE REVIEW

File: MK94.DataGenerator.sln [hunk 1]
```diff
@@ -7,11 +7,6 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MK94.CodeGenerator", "MK94.
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MK94.CodeGenerator.Test", "MK94.CodeGenerator.Test\MK94.CodeGenerator.Test.csproj", "{CEA74623-26F9-4FE4-BADB-D56B54480E08}"
 EndProject
-Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{A9165E45-90B2-40BE-85B7-AF72F88DBAE1}"
```
Comment (mithileshz) [inline MK94.DataGenerator.sln:10]: I don't get why

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                if (typeDef.Properties.Count == 0)
+                    continue;
+
+                var propertiesWithStronglyTypedAttribute = typeDef.Properties.Where(x => x.Info.GetCustomAttributes<StronglyTypedIdAttribute>().Any()).ToList();
+
+                if (propertiesWithStronglyTypedAttribute.Count == 0)
+                    continue;
+
+                var ns = file.Namespace(project.NamespaceResolver(typeDef));
+                IntermediateFileDefinition.IntermediateTypeDefinition type = CreateStronglyTypedIdInterface(ns);
```
Comment (mukut1994) [inline MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs:37]: var

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                {
+                    var stronglyTypedId = ns
+                        .Type(property.Name, MemberFlags.Public)
+                        .WithTypeAsRecord()
+                        .WithTypeAsStruct()
+                        .WithInheritsFrom(CsharpTypeReference.ToRaw("IId"));
+
+                    stronglyTypedId.Property(MemberFlags.Public, CsharpTypeReference.ToType<Guid>(), "Id");
+
+                    property.Type = stronglyTypedId.GetType();
```
Comment (mukut1994) [inline MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs:49]: I'm not sure what this line is supposed to do. I don't like the fact that it's updating the reference.  
This forces modules to run in a specific order, if subsequent modules are supposed to use the new type

**Review by mukut1994 — CHANGES_REQUESTED** [review]

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs (hunk 3 above)
Comment (mithileshz) [reply MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs:49]: this was a typo

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs (hunk 2 above)
Comment (mithileshz) [reply MK94.CodeGenerator/Intermediate/CSharp/Modules/StronglyTypedIdModule.cs:37]: urgh

**Review by mukut1994 — APPROVED** [review]

**Pull request merged**